### PR TITLE
Separeted to source files

### PR DESCRIPTION
**Pull request by aurya-dev**

### CODE REVIEW

File: PlayButton.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "LeakedObjectDetector.h"
+
+struct PlayButton
+{
+    bool isLit;
+    bool isFlash;
+
+    PlayButton () : isLit(false), isFlash(false) {}
```
Comment (matkatmusic) [inline PlayButton.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: RecordButton.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "LeakedObjectDetector.h"
+
+struct RecordButton
+{
+    bool isLit;
+    bool isFlash;
+
+    RecordButton () : isLit(false), isFlash(false) {}
```
Comment (matkatmusic) [inline RecordButton.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: DishWashingProcess.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <iostream>
+
+#include "LeakedObjectDetector.h"
+
+
+struct DishWashingProcess
+{
+    int errorAtMinute;
+    DishWashingProcess(int m) : errorAtMinute(m) {}
```
Comment (matkatmusic) [inline DishWashingProcess.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: Display.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+#include "LeakedObjectDetector.h"
+
+struct Display
+{
+    int numPixelWidth;
+    int numPixelHeight;
+
+    Display(int w = 1024, int h = 678) : numPixelWidth(w), numPixelHeight(h) {}
+    ~Display() { std::cout << "Display distroyed" << std::endl; }
```
Comment (matkatmusic) [inline Display.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: Drumpad.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+struct Drumpad
+{
+    int numPads;
+
+    Drumpad(int num) : numPads(num) {}
+    ~Drumpad() { std::cout << "Drumpad distroyed" << std::endl; }
```
Comment (matkatmusic) [inline Drumpad.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: Pattern.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+#include "LeakedObjectDetector.h"
+
+struct Pattern
+{
+    int playHeadPos;
+    std::string sequenceData = "";
+
+    Pattern( int pos ) : playHeadPos( pos ), sequenceData("") {}
+    ~Pattern() { std::cout << "Pattern distroyed" << std::endl; }
```
Comment (matkatmusic) [inline Pattern.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: VolumeControl.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+#include "LeakedObjectDetector.h"
+
+struct VolumeControl 
+{
+    int maxVolume;
+    int currentVol;
+    bool isMaxVolume();
+    void decreaseMasterVolume(int newVolume);
+
+    VolumeControl(int maxV = 127) : maxVolume(maxV), currentVol(0) {}
```
Comment (matkatmusic) [inline VolumeControl.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: Wrappers.h [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    ServiceStationWrapper(ServiceStation* s) : serStaPtr(s) {}
+    std::unique_ptr<ServiceStation> serStaPtr = nullptr;
+}; 
+
+
+struct ProjectWrapper
+{
+    ProjectWrapper(Project* p) : prjPtr(p) {}
+    std::unique_ptr<Project> prjPtr = nullptr;
+};
```
Comment (matkatmusic) [inline Wrappers.h]: The first time you submitted Project5.3 I should have caught you using `std::unique_ptr`
While it is not wrong to use it (it is the proper tool for holding heap-allocated objects), the video demonstrated how you can manually manage heap-allocated objects yourself.
This is what you needed to do here.

Screen Shot 2021-03-22 at 1 13 56 PM

All of the same requirements regarding NO IN-CLASS IMPLEMENTATIONS ALLOWED must be followed here as well
- [x] revise: implement wrapper classes that demonstrate how to manage the memory themselves.
- [x] revise: NO IN-CLASS IMPLEMENTATIONS ALLOWED

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: Wrappers.h [hunk 9]
```diff
@@ -3,34 +3,69 @@
...
-    std::unique_ptr<DishWasher> diWaPtr = nullptr;
+    DishWasherWrapper(DishWasher* d) 
+    {
+        diWaPtr = d;
+    }
+    ~DishWasherWrapper()
+    {
+        delete diWaPtr;
+    }
+    DishWasher* diWaPtr = nullptr;
```
Comment (matkatmusic) [inline Wrappers.h]:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
```
- [x] revise

File: Wrappers.h [hunk 10]
```diff
@@ -3,34 +3,69 @@
...
-    std::unique_ptr<DishWashingProcess> washPtr = nullptr;
+    DishWashingProcessWrapper(DishWashingProcess* d) 
+    {
+        washPtr = d; 
+    }
+    ~DishWashingProcessWrapper()
+    {
+        delete washPtr;
+    }
+    DishWashingProcess* washPtr = nullptr;
```
Comment (matkatmusic) [inline Wrappers.h]:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
```
- [x] revise

File: Wrappers.h [hunk 11]
```diff
@@ -3,34 +3,69 @@
...
-    std::unique_ptr<DrumMachine> drMaPtr = nullptr;
+    DrumMachineWrapper(DrumMachine* d) 
+    {
+        drMaPtr = d; 
+    }
+    ~DrumMachineWrapper()
+    {
+        delete drMaPtr;
+    }
+    DrumMachine* drMaPtr = nullptr;
```
Comment (matkatmusic) [inline Wrappers.h]:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
```
- [x] revise

File: Wrappers.h [hunk 12]
```diff
@@ -3,34 +3,69 @@
...
-    std::unique_ptr<ServiceStation> serStaPtr = nullptr;
+    ServiceStationWrapper(ServiceStation* s) 
+    {
+        serStaPtr = s; 
+    }
+    ~ServiceStationWrapper()
+    {
+        delete serStaPtr;
+    }
+    ServiceStation* serStaPtr = nullptr;
```
Comment (matkatmusic) [inline Wrappers.h]:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
```
- [x] revise

File: Wrappers.h [hunk 13]
```diff
@@ -3,34 +3,69 @@
...
-    std::unique_ptr<Project> prjPtr = nullptr;
+    ProjectWrapper(Project* p) 
+    {
+        prjPtr = p; 
+    }
+    ~ProjectWrapper()
+    {
+        delete prjPtr;
+    }
+    Project* prjPtr = nullptr;
```
Comment (matkatmusic) [inline Wrappers.h]:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
```
- [x] revise

File: Drumpad.h [hunk 14]
```diff
@@ -4,7 +4,9 @@ struct Drumpad
 {
     int numPads;
 
-    Drumpad(int num) : numPads(num) {}
+    Drumpad(int num) {
+        numPads = num;
+    }
```
Comment (matkatmusic) [inline Drumpad.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: Display.h [hunk 15]
```diff
@@ -7,7 +7,10 @@ struct Display
     int numPixelWidth;
     int numPixelHeight;
 
-    Display(int w = 1024, int h = 678) : numPixelWidth(w), numPixelHeight(h) {}
+    Display(int w = 1024, int h = 678) {
+        numPixelWidth = w; 
+        numPixelHeight = h; 
+    }
```
Comment (matkatmusic) [inline Display.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

File: DishWashingProcess.h [hunk 16]
```diff
@@ -8,7 +8,10 @@
 struct DishWashingProcess
 {
     int errorAtMinute;
-    DishWashingProcess(int m) : errorAtMinute(m) {} 
+    DishWashingProcess(int m) 
+    {
+        errorAtMinute = m; 
+    }
```
Comment (matkatmusic) [inline DishWashingProcess.h]:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: Wrappers.h [hunk 17]
```diff
@@ -1,71 +1,48 @@
 #pragma once
 
+#include "DishWasher.h"
+#include "DishWashingProcess.h"
+#include "DrumMachine.h"
+#include "ServiceStation.h"
+#include "Project.h"
+
```
Comment (matkatmusic) [inline Wrappers.h]: If you're not using anything from these files **in this header file**, `forward declare` here.
then `#include` in the CPP files.

DON'T INCLUDE WHAT YOU DON'T USE

- [x] revise

File: Drumpad.cpp [hunk 18]
```diff
@@ -0,0 +1,10 @@
+#include "Drumpad.h"
+
+Drumpad::Drumpad(int num) 
+{
+    numPads = num;
+}
+
+Drumpad::~Drumpad() { 
+    std::cout << "Drumpad distroyed" << std::endl; 
+}
```
Comment (matkatmusic) [inline Drumpad.cpp]: Curly braces go on their own line.
Follow the course coding standard.
- [x] revise

File: Display.cpp [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+
+
+Display::Display(int w, int h) {
+    numPixelWidth = w; 
+    numPixelHeight = h; 
+}
+
+Display::~Display() { 
+    std::cout << "Display distroyed" << std::endl; 
+}
```
Comment (matkatmusic) [inline Display.cpp]: Curly braces go on their own line.
Follow the course coding standard.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: Wrappers.h (hunk 17 above)
Comment (aurya-dev) [reply Wrappers.h]: Ah ok. I thought the `#include` is generally in the header file. Thanks for the hint!

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**